core/app/models/spree/credit_card.rb

```diff
@@ -46,12 +46,15 @@ def self.default
 
     def default
       Spree::Deprecation.warn("CreditCard.default is deprecated. Please use user.wallet.default_wallet_payment_source instead.", caller)
-      user.wallet.default_wallet_payment_source.payment_source == self
+      return false if user.nil?
+      user.wallet.default_wallet_payment_source.try!(:payment_source) == self
     end
 
     def default=(set_as_default)
       Spree::Deprecation.warn("CreditCard.default= is deprecated. Please use user.wallet.default_wallet_payment_source= instead.", caller)
-      if set_as_default # setting this card as default
+      if user.nil?
```

Comment:
@cbrunsdon that code you referenced (and the `assign_default_credit_card` method that calls it) is a no-op when `order.user` is nil, right?  Is that what you're asking?